synapse/lib/types.py

```diff
@@ -424,8 +424,23 @@ def extend(self, name, opts, info, skipinit=False):
             (synapse.types.Type): A new sub-type instance.
         '''
         tifo = self.info.copy()
+
+        # handle virts by merging them...
+        virts = tifo.get('virts')
+
         tifo.update(info)
 
+        if virts is not None:
+
+            # inherit any virts from our parent type
+            for vname, (tdef, info) in virts.items():
+
+                tifo['virts'].setdefault(vname, (tdef, info))
```

Comment:
I don't think there's a way for this to actually set the value since we pulled `virts` from `tifo['virts']` and don't remove anything